### DISCUSSION TITLE
Unable to login on fresh install - Redis error

### DISCUSSION THREAD

**Discussion by tjfayaz120** [LiveHelperChat/livehelperchat#2028] [Q&A]
Hello we used LHC for many years but recently it seems to have stopped working and we cannot get it to run on any of our sites. We get the same error - after install when trying to login the login page does nothing. The php error is as below. The local redis server is running fine and the php-redis is available and working.

How can we get past this error? Can we somehow tell it not to try to use redis? We tried taking the redis options out of the config file but still get the same error. Also the tmp folder mentioned has full write perms.

`PHP message: PHP Warning:  session_start(): Redis connection not available in /var/www/clients/client1/web1/web/livehelp/ezcomponents/Authentication/src/session/authentication_session.php on line 213PHP message: PHP Warning:  session_start(): Failed to read session data: redis (path: /var/www/clients/client1/web1/tmp) in /var/www/clients/client1/web1/web/livehelp/ezcomponents/Authentication/src/session/authentication_session.php on line 213PHP message: PHP Warning:  session_start(): Redis connection not available in /var/www/clients/client1/web1/web/livehelp/ezcomponents/Authentication/src/session/authentication_session.php on line 213PHP message: PHP Warning:  session_start(): Failed to read session data: redis (path: /var/www/clients/client1/web1/tmp) in /var/www/clients/client1/web1/web/livehelp/ezcomponents/Authentication/src/session/authentication_session.php on line 213" while reading response header from upstream, client: myIP, server: mysite.co.uk, request: "GET /livehelp/site_admin/user/login HTTP/2.0", upstream: "fastcgi://unix:/var/lib/php7.4-fpm/web1.sock:", host: "mysite.co.uk", referrer: "https://mysite.co.uk/livehelp/site_admin/user/login`

**Comment by remdex** [comment]
your redis instance is dead...

**Reply by tjfayaz120** [reply]
> your redis instance is dead...

That was my first thought but it's not. Redis is running fine and in use by the main website. In fact when I watch debug logs on redis it doesn't even receive any connection attempts from lhc

**Comment by remdex** [comment]
because it's php sessions which are using sessions are having trouble...

**Reply by tjfayaz120** [reply]
Any idea how I can disable lhc from using redis? Regardless of if I remove the redis from the config it's still has the same error. 

Also interested to know why won't it even try to connect to redis at all. Nothing hits redis.

**Reply by remdex** [reply]
really, I have nothing todo with your env... it's php-fpm settings...

**Reply by tjfayaz120** [reply]
Thanks I'll try some more testing and see if I can work it out before we give up and use a different one.

I'm just wondering why we cannot turn off the redis on lhc? I will try an older pre-redis version 1.29 ...

**Reply by remdex** [reply]
who told you can't? it's env problem. lhc knows nothing about redis... Seems the only problem here is your lack of basic knowledge...

**Reply by tjfayaz** [reply]
OK this is now resolved by changing the default session handler in fpm to files instead of redis - strangely this works fine for all other sites but not LHC itself. 

So indeed the issue appears to be outside of LHC however it's not correct to say LHC knows nothing about redis and I find your reply rather rude. LHC is configured to use redis as per the config file of LHC so it DOES know something about redis, even if that's just a part of the config that it may or may not use in this version. 

settings/settings.ini.php you have this segment

` 'redis' =>
    array (
      'server' =>
      array (
        'host' => 'localhost',
        'port' => 6379,
        'auth' => NULL,
        'database' => 0,
      ),
    ),`

I feel like there is some reason LHC cannot use redis as a session handler and will investigate this further but at least it's working for now.

**Reply by remdex** [reply]
It does knows nothing about it. Settings file contains redis section, but is not used anywhere. So your assumption is incorrect.

In general even error tells you that it's php error if you just google'd it. Instead you decided to post unrelated thing to lhc itself and just take time from project itself...

But well... I can't teach everybody just use google 😥
